dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2625,190 @@ void FileSprayer::pushParts()
+                {
+                    auto response = &f.Poll();
+                    auto props = f.Value();
+                    auto status = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Shares::Models::CopyStatus::Pending);
+                    curprogress.outputLength = props.FileSize;
+                    curprogress.resultTime.setString(props.LastModified.ToString().c_str());
+                    if (status==Shares::Models::CopyStatus::Success)
+                    {
+                        curprogress.status = OutputProgress::StatusCopied;
+                        curprogress.outputCRC = curprogress.inputCRC;
+                    }
+                    else if (status==Shares::Models::CopyStatus::Pending)
+                    {
+                        curprogress.status =  OutputProgress::StatusActive;
+                        anyRemaining=true;
+                    }
+                    else
+                        return false; //Aborted or Failed -> return false to use older copy method
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("transferUsingAPI .poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+                    return false;
+                }
+                if (isAborting())
+                    break;
+            }
+        }
+        else
+        {
+            for(auto & b: blobCopyOp)
+            {
+                OutputProgress & curprogress = progress.item(*itProgress);
+                ++itProgress;
+                if (curprogress.status==OutputProgress::StatusCopied)
+                    continue;
+                try
+                {
+                    auto response = &b.Poll();
+                    auto props = b.Value();
+                    auto status = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Blobs::Models::CopyStatus::Pending);
+                    curprogress.outputLength = props.BlobSize;
+                    curprogress.resultTime.setString(props.LastModified.ToString().c_str());
+                    if (status==Blobs::Models::CopyStatus::Success)
+                    {
+                        curprogress.status = OutputProgress::StatusCopied;
+                        curprogress.outputCRC = curprogress.inputCRC;
+                    }
+                    else if (status==Blobs::Models::CopyStatus::Pending)
+                    {
+                        curprogress.status =  OutputProgress::StatusActive;
+                        anyRemaining=true;
+                    }
+                    else
+                        return false; //Aborted or Failed -> return false to use older copy method
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("transferUsingAPI .poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
```

Comment:
Not sure this should abort before the other copies have completed - see similar comment above.